### DISCUSSION TITLE
vcpkg and OS going to sleep for idle mode detected.

### DISCUSSION THREAD

**Discussion by Raffaello** [microsoft/vcpkg#17825] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
On widows if there are no user input for a while the system is detected on idle and go to sleep.

`vcpkg` doesn't make the OS aware that is busy. So resulting in the system suspend if left just working install few long time consuming library.

**Proposed solution**
set the vcpkg thread, at least for windows system, to keep the system awake while installing/compiling libraries.
like reported here :
[SetThreadExecutionState](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setthreadexecutionstate)

**Describe alternatives you've considered**
well, tried simple tools that rely on the function above but Win10 seems ignoring too.
So doing in chunks install, smaller batch of library install.

**Additional context**
It is not really a high priority feature, but would be nice to make the tool "more complete" and reliable.
thanks.

**Comment by ghost** [comment]
It would be a great addition
- does anyone know if there a comparable feature on all platforms besides windows? 
- the linked article is discussing the feature in the `winbase.h` header file for windows.

**Reply by Raffaello** [reply]
> It would be a great addition
> 
> * does anyone know if there a comparable feature on all platforms besides windows?
> * the linked article is discussing the feature in the `winbase.h` header file for windows.

I am not sure, but in Linux i think is working differently than from Windows, and there are kernel parameters that allowing different behaviors: https://www.kernel.org/doc/html/v4.19/admin-guide/pm/sleep-states.html

----

also the idle time is GUI dependent, for example with GNOME is just any user input events (key, mouse,..) to reset IDLE_TIME
https://blogs.gnome.org/hughsie/2009/08/17/gnome-power-manager-and-blanking-removal-of-bodges/

----

but for GNOME linux could be something like this: http://manpages.ubuntu.com/manpages/cosmic/man1/gnome-session-inhibit.1.html

----

for mac should be something equivalent to as there is an app, caffeineted, to achieve that, https://apps.apple.com/us/app/caffeinated-anti-sleep-app/id1362171212?mt=12

**Comment by Raffaello** [comment]
it is enough to override the power setting in some circumstancing of long compilation of some libraries and that's it, that's why low priority, but there could be other cases when. for absurd, an enterprise laptop as power management policy that cannot be changed and go to sleep after some minutes of inactivity, idle mode for instance, and with modern CPU, even when 1 core is processing, the OS speculate the system is on idle mode, due to a threshold of resource usage probably... so don't really won't to bring as well that the OS idle detection function could be evovled into a "smart idle detection" function as it is pointless when is enough that the program is aware and communicating to the OS that is doing work... :)

**Reply by dg0yt** [reply]
> an enterprise laptop

On a laptop, you might try enabling "Presentation mode" (from the "Mobility Center" or via `presentationsettings /start`).
